From 7ac56879150a01a02dda2cda2c9eccaf0ea8f872 Mon Sep 17 00:00:00 2001
From: Suraj Jaiswal <quic_jsuraj@quicinc.com>
Date: Thu, 9 Jan 2025 17:07:00 +0530
Subject: [PATCH] net: phy: AQR phy 10M fix

With 10Mbps mac link speed is not properly set.
fix 10M.

Change-Id: I5a1ce089cc03594813b98ac4a4cae26c541733ca
Signed-off-by: Suraj Jaiswal <quic_jsuraj@quicinc.com>
---
 drivers/net/phy/aquantia/aquantia_main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/phy/aquantia/aquantia_main.c b/drivers/net/phy/aquantia/aquantia_main.c
index 169df8f04289..98291832807d 100644
--- a/drivers/net/phy/aquantia/aquantia_main.c
+++ b/drivers/net/phy/aquantia/aquantia_main.c
@@ -323,7 +323,7 @@ static int aqr107_read_rate(struct phy_device *phydev)
 		return val;
 
 	if (FIELD_GET(VEND1_GLOBAL_CFG_RATE_ADAPT, val) ==
-	    VEND1_GLOBAL_CFG_RATE_ADAPT_PAUSE)
+	    VEND1_GLOBAL_CFG_RATE_ADAPT_PAUSE && phydev->speed != SPEED_10)
 		phydev->rate_matching = RATE_MATCH_PAUSE;
 	else
 		phydev->rate_matching = RATE_MATCH_NONE;
-- 
2.25.1

